polyphemus/lib/etls/redcap/redcap_etl_script_runner.rb

```diff
@@ -0,0 +1,83 @@
+require_relative 'lib/client'
+require_relative 'lib/model'
+require_relative 'lib/loader'
+require_relative 'lib/script'
+require_relative 'lib/value'
+require_relative 'lib/form'
+require_relative 'lib/project'
+
+require_relative '../etl_script_runner'
+require_relative '../../magma_record_etl'
+
+class Polyphemus
+  class RedcapEtlScriptRunner < EtlScriptRunner
+
+    attr_reader :magma_client, :update_request, :model_names, :redcap_tokens, :redcap_host, :magma_host, :dateshift_salt
+
+    def initialize(project_name:, model_names:, redcap_tokens:, redcap_host:, magma_host:, dateshift_salt:)
+      # Override initialize, user won't be passing in a filename directly.
+      raise "Must provide at least one REDCap token." unless redcap_tokens
+
+      @file_path = File.join(File.dirname(__FILE__), 'projects', "#{project_name}.rb")
+
+      raise "Project configuration does not exist." unless File.file?(@file_path)
+
+      @project_name = project_name
+      @model_names = model_names
+      @redcap_tokens = redcap_tokens
+
+      raise "REDCap host must use https://" if redcap_host.start_with?("http://")
+      raise "Magma host must use https://" if magma_host.start_with?("http://")
+
+      @redcap_host = redcap_host
+      @magma_host = magma_host
+      @dateshift_salt = dateshift_salt
+
+      warn "Warning: no dateshift_salt provided, will use a randomly generated one." unless dateshift_salt
```

Comment:
Is this something that we might want to fail with error on instead, so that users aren't surprised by dates if something isn't configured right?  Just not sure either way.

Reply:
That's a good point, I'm not sure what the expected behavior is during analysis (since the date differences should still be the same with different salts). However, it does seem confusing if dates changed between loading events (due to a salt changing), so throwing an error makes sense. Thanks!

Reply:
Yeah a consistent salt is vital. I would eventually like to do this in magma's date_time attribute via a loading hook, to avoid this issue. Really this is a valuable secret guarding PHI and should be tightly controlled. In the meanwhile ensuring we all use the same salt is required so the dates are shifted consistently (otherwise they are useless).